File: academic/octave/py_octave/data_load.py
import numpy as np
#import scipy.io as sio
from oct2py import octave as oc
import matplotlib.pyplot as plt

class DataLoad:
    def __init__(self,n_start,n_stop):
        #input_data=sio.loadmat("data_input.mat")['input_data']
        #output_data=sio.loadmat("data_output.mat")['output_data']
        input_data= oc.data_input
        output_data= oc.data_output
        
        u_11 = input_data[n_start:n_stop,1]
        u_12 = input_data[n_start:n_stop,3]
        u_13 = input_data[n_start:n_stop,2]
        
        y_11 = output_data[n_start:n_stop,1]
        y_12 = output_data[n_start:n_stop,2]
                
        self.u11 = u_11/u_11.max()
        self.u12 = u_12/u_12.max()
        self.u13 = u_13/u_13.max()
        self.y11 = y_11/y_11.max()
        self.y12 = y_12/y_12.max()
        
        self.t = np.arange(0,n_stop-n_start,1)
        self.dim = np.shape(self.t)[0]
        
    def plot(self):
        plt.figure()
        plt.title('Measured')
        plt.subplot(5, 1, 1)
        plt.plot(self.u11)
        plt.ylabel('Input 1')
        plt.subplot(5, 1, 2)
        plt.plot(self.u12)
        plt.ylabel('Input 2')
        plt.subplot(5, 1, 3)
        plt.plot(self.u13)
        plt.ylabel('Input 3')
        plt.subplot(5, 1, 4)
        plt.plot(self.y11)
        plt.ylabel('Output 1')
        plt.subplot(5, 1, 5)
        plt.plot(self.y12)
        plt.ylabel('Output 2')
        
        plt.xlabel('Data Series')
        plt.show()

File: academic/octave/py_octave/data_load.m
% https://www.mathworks.com/matlabcentral/newsreader/view_thread/33101

% bersihkan segala-galanya
close all
clc
clear all

% masukkan data ke memori
data_input;
data_output;

% tentukan jangkauan data
% dimulai dari 20 sehingga tidak perlu mengambil 20 nilai awal yang masih berantakan
jangkauan = (20:1450);

% parse (memisahkan) data untuk tiap variable input
u_11 = input_data(jangkauan,2); % input #1 di array ke 2
u_12 = input_data(jangkauan,4); % input #2 di array ke 4
u_13 = input_data(jangkauan,3); % input #3 di array ke 3

% parse (memisahkan) data untuk tiap variable output
y_11 = output_data(jangkauan,2); % output #1 di array ke 2
y_12 = output_data(jangkauan,3); % output #2 di array ke 3

% normalisasi nilai pada rentang 0 s/d 1
u11 = u_11/max(u_11);
u12 = u_12/max(u_12);
u13 = u_13/max(u_13);
y11 = y_11/max(y_11);
y12 = y_12/max(y_12);

% Plotting data
figure('Name','Input ter-Normalisasi','NumberTitle','off')
subplot(511),plot(u11);
grid;
legend('input #1');
subplot(512),plot(u12);
grid;
legend('input #2');
subplot(513),plot(u13);
grid;
legend('input #3');
subplot(514),plot(y11);
grid;
legend('output #1');
subplot(515),plot(y12);
grid;
legend('output #2');
